src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectOrCollectionInitializerExpression.cs

```diff
@@ -276,8 +276,9 @@ private BoundExpression VisitObjectInitializerMember(BoundObjectInitializerMembe
 
             BoundExpression? rewrittenLeft = null;
 
-            // Do not lower pointer access yet, we'll do it later.
-            if (assignment.Left.Kind != BoundKind.PointerElementAccess)
+            // Do not lower pointer or implicit indexer access yet, we'll do them later.
+            BoundKind leftKind = assignment.Left.Kind;
+            if (leftKind != BoundKind.PointerElementAccess && leftKind != BoundKind.ImplicitIndexerAccess)
```

Comment:
> && leftKind != BoundKind.ImplicitIndexerAccess [](http://example.com/codeflow?start=58&length=47)

It is not obvious why this is necessary. Why regular way of doing things works for regular indexer and doesn't work for an implicit indexer? It is quite possible that I would be more comfortable with adjusting existing code paths instead of introducing completely new case below.

Reply:
The regular indexer case is handled inside `VisitObjectInitializerMember` which takes a `BoundObjectInitializerMember`. So it doesn't seem very suitable for sharing.

I could convert the ternary expression below into a switch and essentially do `rewrittenLeft = VisitImplicitIndexerAccess((BoundImplicitIndexerAccess)assignment.Left, ...);` here (instead of `rewrittenAccess = VisitImplicitIndexerAccess(...)` below) if that feels more comfortable, but I don't think we'll be able to share.